src/validator.cc

```diff
@@ -1080,35 +1080,35 @@ void Validator::CheckAssertReturnNanCommand(const Action* action) {
 void Validator::CheckCommand(const Command* command) {
```

Comment:
True. I like keeping the validation logic in validator.cc, though. If we went this route, I think I'd prefer adding visitors for these to prevent having to cast all the time.